torch/csrc/jit/codegen/cuda/transform_replay.cpp

```diff
@@ -718,17 +718,38 @@ bool TransformPropagator::replayPasC(
   }
 
   auto pairwiseMap = PairwiseRootDomainMap(producer_tv, consumer_tv);
-  auto producerAsC = TransformReplay::replayPasC(
+  auto replayed_producer = TransformReplay::replayPasC(
       producer_tv, consumer_tv, consumer_pos_it->second, pairwiseMap);
 
+  auto producer_root = producer_tv->getMaybeRFactorDomain();
+  auto replayed_domain = replayed_producer.first->domain();
+
+  // Find the number of root IDs involved in the transformation
+  auto dep_vals = DependencyCheck::getAllValsBetween(
+      {producer_root.begin(), producer_root.end()},
+      {replayed_domain.begin(),
+       replayed_domain.begin() + replayed_producer.second});
+
+  std::unordered_set<Val*> dep_vals_set{dep_vals.begin(), dep_vals.end()};
+
+  auto n_transformed_root_dims = std::count_if(
+      producer_root.begin(),
+      producer_root.end(),
+      [&dep_vals_set](IterDomain* root_id) {
+        return dep_vals_set.find(root_id) != dep_vals_set.end();
+      });
+
   if (replayed_pos.find(producer_tv) != replayed_pos.end()) {
-    if (producerAsC.second <= replayed_pos.at(producer_tv)) {
+    if (n_transformed_root_dims <= n_replayed_root_dims.at(producer_tv) ||
+        (n_transformed_root_dims == n_replayed_root_dims.at(producer_tv) &&
+         replayed_producer.second <= replayed_pos.at(producer_tv))) {
```

Comment:
If I'm reading this correctly, the rhs of `||` seems redundant. To be true, `n_transformed_root_dims == n_replayed_root_dims.at(producer_tv)` must hold, but then that's covered by the lhs condition of `||`.

Reply:
Do you have an example fusion for which this change would fix the propagation?

Reply:
Suppose `n_transformed_root_dims == n_replayed_root_dims.at(producer_tv)`. Wouldn't we want to return true when `replayed_producer.second > replayed_pos.at(producer_tv)`?

Reply:
I think this should be:
```
n_transformed_root_dims < n_replayed_root_dims.at(producer_tv) ||
        (n_transformed_root_dims == n_replayed_root_dims.at(producer_tv) &&
         replayed_producer.second <= replayed_pos.at(producer_tv))
```
Just forgot to change the first section without the second. I don't know if it's necessary, but seemed harmless.